pkg/operator/ceph/csi/spec.go

```diff
@@ -676,12 +676,29 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 	return nil
 }
 
-func (r *ReconcileCSI) stopDrivers(ver *version.Info) error {
+// return false is rook operator is not the owner of csi driver,
+// so csi driver do not get deleted as it isn't deployed by
+// rook operator and someone externally installed it
+func (r *ReconcileCSI) validateDriverOwner(ownerInfo *k8sutil.OwnerInfo, deploymentName string) bool {
+	getDeployment, err := r.context.Clientset.AppsV1().Deployments(r.opConfig.OperatorNamespace).Get(r.opManagerContext, deploymentName, metav1.GetOptions{})
+	if err != nil {
+		logger.Warning("csi deployment pod not present")
```

Comment:
return here or else below code will panic

Reply:
I am returning `return true` at the end of the function, so the original code executes as it is if the deployment even not exit.

Reply:
log actual error in the Warning or log it as an error other than NotFound

Reply:
please check the new flow